### ISSUE TITLE
[Sheet, Shell Panel] Automatically render Panel

### ISSUE THREAD

**Issue by SkyeSeitz** [Esri/calcite-design-system#9828]
**monday.com sync:** #9583322778

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Explore setting up Calcite Sheet and Shell Panel to automatically render a Panel as we are implementing with [Dialog](https://github.com/Esri/calcite-design-system/issues/7886). 

### Acceptance Criteria

[Link to full Figma specs 🔗](https://www.figma.com/design/Uw57dyzHngOkVaSOfg0gUj/-Sheet--Shell-Panel--Popover--Automatically-render-Panel?node-id=301-305&p=f&t=hsuPDXyQSq9Vl8IG-11)

**Shell Panel**
**Props passed through from Panel**
- scale
- heading
- description
- closable
- collapsible
- collapsed
- loading
- menu-open

**Slots passed through from Panel**
- header-actions-start
- header-content
- header-actions-end
- header-menu-actions
- action-bar
- content-top
- default slot
- content-bottom
- fab
- footer-start
- footer-end

**Sheet**
**Panel background color** 
To match the behavior of Panel rendered in Dialog, the panel background color changes from its default `color.background` to `color.foreground.1`

**Props passed through from Panel**
- scale
- heading
- description
- closable
- collapsible
- collapsed
- loading
- menu-open
- Slots passed through from Panel
- header-actions-start
- header-content
- header-actions-end
- header-menu-actions
- action-bar
- content-top
- default slot
- content-bottom
- fab
- footer-start
- footer-end

### Which Component

Sheet, Shell Panel

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (design)

**Comment by driskull** [comment]
I have a draft of this one for whoever wants to take it in Nov/Dec: https://github.com/Esri/calcite-design-system/pull/13151

The draft excludes popover because that one still needs a design on how it will work with panel and its close affordance. Panel would need an option to have the close button embedded within its content. cc @SkyeSeitz

**Comment by SkyeSeitz** [comment]
Thanks, @driskull! 
@geospatialem Should this one go into design before devs can move forward with Popover?

**Comment by driskull** [comment]
@SkyeSeitz  yes, it should go through design first.

**Comment by geospatialem** [comment]
Added to the November/December milestone for design, once the scope is complete across components we should add to the January milestone for development work.

cc @SkyeSeitz @driskull

**Comment by DintaMel** [comment]
Separated popover to a new issue. #13989
